test/new-e2e/tests/windows/service-test/startstop_test.go

```diff
@@ -0,0 +1,335 @@
+				s.T().Logf("Errors and warnings from %s event log:\n%s", logName, out)
+			}
+		}
+		// collect agent logs
+		s.collectAgentLogs()
+	}
+}
+
+func (s *startStopTestSuite) collectAgentLogs() {
+	host := s.Env().RemoteHost
+	outputDir, err := runner.GetTestOutputDir(runner.GetProfile(), s.T())
+	if err != nil {
+		s.T().Fatalf("should get output dir")
+	}
+
+	s.T().Logf("Collecting agent logs")
+	logsFolder, err := host.GetLogsFolder()
+	if !s.Assert().NoError(err, "should get logs folder") {
+		return
+	}
+	entries, err := host.ReadDir(logsFolder)
+	if !s.Assert().NoError(err, "should read log folder") {
+		return
+	}
+	for _, entry := range entries {
+		s.T().Logf("Found log file: %s", entry.Name())
+		err = host.GetFile(
+			filepath.Join(logsFolder, entry.Name()),
+			filepath.Join(outputDir, entry.Name()),
+		)
+		s.Assert().NoError(err, "should download %s", entry.Name())
+	}
+}
+
+func (s *startStopTestSuite) startAgent() {
+	host := s.Env().RemoteHost
+	err := windowsCommon.StartService(host, "datadogagent")
+	s.Require().NoError(err, "should start the datadogagent service")
+}
+
+func (s *startStopTestSuite) requireAllServicesState(expected string) {
+	// ensure all services are running
+	s.assertAllServicesState(expected)
+
+	if s.T().Failed() {
+		// stop test if not all services are running
+		s.FailNow("not all services are %s", expected)
+	}
+}
+
+func (s *startStopTestSuite) assertAllServicesState(expected string) {
+	host := s.Env().RemoteHost
+
+	for _, serviceName := range s.expectedInstalledServices() {
+		s.Assert().EventuallyWithT(func(c *assert.CollectT) {
+			status, err := windowsCommon.GetServiceStatus(host, serviceName)
+			require.NoError(c, err)
+			if !assert.Equal(c, expected, status, "%s should be %s", serviceName, expected) {
+				s.T().Logf("waiting for %s to be %s, status %s", serviceName, expected, status)
+			}
+		}, 1*time.Minute, 1*time.Second, "%s should be in the expected state", serviceName)
+	}
+}
+
+func (s *startStopTestSuite) stopAllServices() {
+	host := s.Env().RemoteHost
+
+	// stop agent first, it should stop all services
+	s.T().Logf("Stopping the agent service...")
+	err := windowsCommon.StopService(host, "datadogagent")
+	s.Require().NoError(err, "should stop the datadogagent service")
+	s.T().Logf("Agent service stopped")
+
+	// ensure all services are stopped
+	for _, serviceName := range s.expectedInstalledServices() {
+		s.Assert().EventuallyWithT(func(c *assert.CollectT) {
+			status, err := windowsCommon.GetServiceStatus(host, serviceName)
+			require.NoError(c, err)
+			if !assert.Equal(c, "Stopped", status, "%s should be stopped", serviceName) {
+				s.T().Logf("%s still running, sending stop cmd", serviceName)
+				err := windowsCommon.StopService(host, serviceName)
+				assert.NoError(c, err, "should stop %s", serviceName)
+			}
+		}, 1*time.Minute, 1*time.Second, "%s should be in the expected state", serviceName)
+	}
```

Comment:
Should we also use agent start/stop/restart commands?

Reply:
added